internal/pagefile_new.go

```diff
@@ -15,11 +15,17 @@ import (
 	"os"
 )
 
+type ReadWriterAt interface {
+	io.ReaderAt
+	io.WriterAt
+	Stat() (os.FileInfo, error)
+	Name() string
+}
+
 // RestoreMissingPages restores missing pages (zero blocks)
 // of local file with their base backup version
-func RestoreMissingPages(base io.Reader, file *os.File) error {
+func RestoreMissingPages(base io.Reader, file ReadWriterAt) error {
```

Comment:
"file" is not a file anymore. Rename it in all functions with such inconvenient naming.

Reply:
Renamed in 2e4979b381e5d441acaa7269833595e39f651f7f